core/src/main/java/org/apache/cloudstack/storage/to/VolumeObjectTO.java

```diff
@@ -343,6 +344,10 @@ public void setDeviceId(Long deviceId) {
     }
 
     public void setCacheMode(DiskCacheMode cacheMode) {
+        if (DiskCacheMode.HYPERVISOR_DEFAULT.equals(cacheMode) && !Hypervisor.HypervisorType.KVM.equals(hypervisorType)) {
+            this.cacheMode = DiskOffering.DiskCacheMode.NONE;
```

Comment:
this is a bit confusing. `DiskCacheMode.HYPERVISOR_DEFAULT` ?-> `DiskOffering.DiskCacheMode.NONE`. are these the same enum? and if so, why not keep it as `HYPEVISOR_DEFAULT` and handle it in the background.

Reply:
as I understand, this code block affects other hypervisors , not KVM. 
for KVM, it uses DiskCacheMode.HYPERVISOR_DEFAULT in the backend
for other hypervisors, it uses DiskCacheMode.NONE instead in the backend. I think we should mention it in API param description and documentation

Reply:
yes, but also make clear these are the same enum/value and not accidentally similar named values from different locations. I.e. remove the `DiskOffering` bit:
```suggestion
        if (DiskCacheMode.HYPERVISOR_DEFAULT.equals(cacheMode) && !Hypervisor.HypervisorType.KVM.equals(hypervisorType)) {
            this.cacheMode = DiskCacheMode.NONE;
```